### PR TITLE
Added multiedge keyword to `maximal()` method

### PR DESCRIPTION
**Pull request by nwlandry**
Also removed an inconsistent unit test.

### CODE REVIEW

File: xgi/classes/reportviews.py [hunk 1]
```diff
@@ -693,9 +693,15 @@ def singletons(self):
         """
         return self.filterby("size", 1)
 
-    def maximal(self):
-        """Returns the maximal edges as an EdgeView. Maximal edges are those that are not subsets of any other edges in the hypergraph.
+    def maximal(self, multiedges=True):
```
Comment (leotrs) [inline xgi/classes/reportviews.py:696]: I think the current parameter name and documentation are not terribly descriptive. I suggest  something like `with_multiplicity`, with documentation along the liens of "if a maximal edge occurs multiple times, return it according to its multiplicity". Also add a "Notes" section to the docstring that explains that setting `with_multiplicity` to True or False may have different performances depending on the number of repeated maximal edges.

**Review by leotrs — APPROVED** [review]

File: xgi/classes/reportviews.py (hunk 1 above)
Comment (nwlandry) [reply xgi/classes/reportviews.py:696]: I just want to clarify that this keyword specifies whether multiedges can be maximal, not whether to return an edge with its multiplicity (though I might have misunderstood your point). If False, we are enforcing strict inclusion for maximal hyperedges and therefore, multiedges can't be included. If, however, multiedges=True, then we add all the maximal multiedges with non-strict inclusion.

I agree that it is not clear - any ideas for a new keyword name? I'm not sure `with_multiplicity` exactly captures what it is doing. I'll add a more detailed note like you suggested.

Comment (leotrs) [reply xgi/classes/reportviews.py:696]: Thanks for the explanation. I agree `with_multiplicity` is also not ideal. How about simply `strict`?

Comment (nwlandry) [reply xgi/classes/reportviews.py:696]: Perfect. I'll work on that right now.

**Pull request merged**